server/src/main/java/com/linecorp/centraldogma/server/internal/storage/repository/git/GitRepository.java

```diff
@@ -549,6 +577,90 @@ private Map<String, Entry<?>> blockingFind(
         }
     }
 
+    // TODO(ikhoon): Design a storage format to store the revision information of each file and read them
+    //               without retrieving the history.
+    private Map<String, Revision> blockingFindLastFileRevisions(Revision from, Revision to,
+                                                                List<String> paths) {
+        final RevisionRange range = normalizeNow(from, to);
+        final RevisionRange descendingRange = range.toDescending();
+        final Set<String> remainingFiles = new HashSet<>(paths);
+        final Map<String, Revision> lastRevisionMap = new HashMap<>();
+
+        // At this point, we are sure: from.major >= to.major and read lock is acquired.
+        try (ObjectReader reader = jGitRepository.newObjectReader();
```

Comment:
What do you think of using the caching object reader that is introduced in #1097? 😉